src/clog/ob_log_file_pool.cpp

```diff
@@ -417,19 +419,54 @@ int ObLogWriteFilePool::create_tmp_file(const file_id_t file_id, char* fname, co
 int ObLogWriteFilePool::update_free_quota(const char* path, const int64_t percent, const int64_t limit_percent)
 {
   int ret = OB_SUCCESS;
-  struct statfs fsst;
+  int64_t used_size = 0;
+
   if (NULL == path || 0 > percent || 100 < percent || 0 > limit_percent || 100 < limit_percent) {
     ret = OB_INVALID_ARGUMENT;
-  } else if (OB_UNLIKELY(0 != statfs(path, &fsst))) {
+  } else if (OB_FAIL(get_total_used_size(used_size))) {
+    ret = OB_IO_ERROR;  
+    COMMON_LOG(ERROR, "get_total_used_size fail", K(ret));
+  } else if(OB_FAIL(calculate_free_quota(path, used_size, percent, limit_percent))){
     ret = OB_IO_ERROR;
-    CLOG_LOG(ERROR, "statfs error", K(ret), K(path), K(errno), KERRMSG);
-  } else {
-    const int64_t total_size = (int64_t)fsst.f_bsize * (int64_t)fsst.f_blocks;
-    const int64_t free_quota =
-        (int64_t)fsst.f_bsize * ((int64_t)fsst.f_blocks * percent / 100LL - (int64_t)(fsst.f_blocks - fsst.f_bavail));
-    const int64_t used_size = (int64_t)fsst.f_bsize * (int64_t)(fsst.f_blocks - fsst.f_bavail);
-    int64_t limit_quota = (int64_t)fsst.f_bsize *
-                          ((int64_t)fsst.f_blocks * limit_percent / 100LL - (int64_t)(fsst.f_blocks - fsst.f_bavail));
+    CLOG_LOG(ERROR, "calculate free quota error", K(ret), K(path), K(errno), KERRMSG);
+  } 
+  return ret;
+}
+
+int ObLogWriteFilePool::init_free_quota(const char* path, const int64_t percent, const int64_t limit_percent)
+{
+  int ret = OB_SUCCESS;
+  const int64_t used_size = 0;
+
+  if (NULL == path || 0 > percent || 100 < percent || 0 > limit_percent || 100 < limit_percent) {
+    ret = OB_INVALID_ARGUMENT;
+  } else if(OB_FAIL(calculate_free_quota(path, used_size, percent, limit_percent))){
+    ret = OB_IO_ERROR;
+    CLOG_LOG(ERROR, "calculate free quota error", K(ret), K(path), K(errno), KERRMSG);
+  }
+  return ret;
+}
+
+int ObLogWriteFilePool::calculate_free_quota(const char* path, const int64_t used_size, const int64_t percent, const int64_t limit_percent)
+{
+  int ret = OB_SUCCESS;
+  struct statvfs svfs;
+  const int64_t clog_disk_limit_size = ObServerConfig::get_instance().clog_disk_limit_size;
+  int64_t total_size = 0;
+
+  if (OB_UNLIKELY(0 != statvfs(path, &svfs))) {
+    ret = OB_IO_ERROR;
+    CLOG_LOG(ERROR, "statvfs error", K(ret), K(path), K(errno), KERRMSG);
+  } else{
+    if (clog_disk_limit_size != 0){
+      total_size = clog_disk_limit_size;
```

Comment:
if clog_disk_limit_size is set， use min(clog_disk_limit_size, disk total) as real total_size

Reply:
if clog_disk_limit_size set by user is greater than disk size, observer can't start normally. This check logic is done by  "ObConfigClogDiskLimitSizeChecker::check" (in [src/share/config/ob_config_helper.cpp])